student/tests.py

```diff
@@ -86,22 +141,47 @@ def test_get_students_graduate_after_returns_students_that_graduate_after_the_gi
                 example_student_b.educational_institution.id, example_student_b.graduation_date)]
 
 
-@pytest.mark.django_db
+@ pytest.mark.django_db
 def test_if_user_is_a_student_returns_true_if_the_auth_user_is_associated_with_a_student_account(example_student_a):
-    example_user_not_student = User.objects.create_user("notstudent", "password")
+    example_user_not_student = User.objects.create_user(
+        "notstudent", "password")
     example_user_not_student.save()
     assert Student.is_student(example_user_not_student.id) is False
     assert Student.is_student(example_student_a.user.id) is True
 
 
-@pytest.mark.django_db
+@ pytest.mark.django_db
 def test_get_user_as_student_object(example_student_a):
     assert isinstance(Student.get_student(example_student_a.user.id), Student)
 
 
-@pytest.mark.django_db
+@ pytest.mark.django_db
 def test_get_student_raises_exception_for_non_student_user():
-    example_user_not_student = User.objects.create_user("notstudent", "password")
+    example_user_not_student = User.objects.create_user(
+        "notstudent", "password")
     example_user_not_student.save()
     with pytest.raises(Student.DoesNotExist):
         Student.get_student(example_user_not_student.id)
+
+
+@pytest.mark.django_db
+def test_sign_up_valid_student(valid_student_data):
+    form = StudentRegistrationForm(data=valid_student_data)
+    if form.is_valid():
+        user = form.save()
+        assert User.objects.filter(pk=user.user.id).exists()
+        assert Student.objects.filter(pk=user).exists()
+    else:
+        assert False
+
+
+@pytest.mark.django_db
+def test_sign_up_invalid_student(invalid_student_data):
+    for user_data in invalid_student_data:
+        invalid = False
+        form = StudentRegistrationForm(data=user_data)
+        try:
+            form.save()
+        except ValueError:
+            invalid = True
+        assert invalid
```

Comment:
I'm seeing tests for the for object but not for the views. What guarantee do we have that the views are actually using that object?

The right way to test a for as through the view via `client`. That way you can test both the validation logic provided by the form and the control/routing logic provided by the view.

Reply:
will be fixed. I think this is the same comment on both of the text files. (recruiter and student)